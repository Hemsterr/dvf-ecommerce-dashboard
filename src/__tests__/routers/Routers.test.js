// Libs
import React from 'react'
import { render, wait } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

// Component
import Routers from '../../routers'
import AppContext from '../../contexts'
import ROUTES from '../../constants/routes'

const store = {
  dispatch: jest.fn(),
  state: {
    alterations: [],
  },
}

const FittingOptionsScreen = router => (
  <AppContext.Provider value={store}>
    <MemoryRouter initialEntries={[router]}>
      <Routers />
    </MemoryRouter>
  </AppContext.Provider>
)

describe('Routers screen', () => {
  const { container, rerender } = render(FittingOptionsScreen(ROUTES.HOME))
  test('Testing render default router', async () => {
    await wait()
    expect(container).toMatchSnapshot()
    expect(container.querySelectorAll('.landing')).toBeTruthy()
  })

  test('Testing render order alteration router', () => {
    rerender(FittingOptionsScreen(ROUTES.BOOKING))

    expect(container.querySelectorAll('.fitting-options')).toBeTruthy()
  })

  test('Testing render order alteration router', () => {
    rerender(FittingOptionsScreen(ROUTES.ORDER_ALTERATIONS))

    expect(container.querySelectorAll('.orders')).toBeTruthy()
  })
})
