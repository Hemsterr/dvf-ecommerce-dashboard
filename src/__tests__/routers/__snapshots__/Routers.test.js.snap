// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routers screen Testing render default router 1`] = `
<div>
  <div>
    <div
      class="landing"
    >
      <div>
        <h1
          class="landing__hemster-logo"
        >
          <img
            alt="Hemster"
            src="[object Object]"
          />
        </h1>
        <p
          class="landing__header"
        >
          alterations
        </p>
        <h1
          class="landing__hemster-logo"
        >
          <img
            alt="DVF"
            src="[object Object]"
          />
        </h1>
        <p
          class="landing__title"
        >
          Please enter your order number:
        </p>
        <p
          class="landing__description"
        >
          (order # can be found in your confirmation email)
        </p>
        <div
          class="landing__input col-lg-4 col-xs-12 col-sm-12 input-group"
        >
          
          <div>
            <input
              class="input-group__text  input-group__without-border"
              placeholder=""
              type="text"
              value=""
            />
            
          </div>
        </div>
        <p
          class="landing__email"
        >
          Please enter your email:
        </p>
        <div
          class="landing__input col-lg-4 col-xs-12 col-sm-12 input-group"
        >
          
          <div>
            <input
              class="input-group__text  input-group__without-border"
              placeholder=""
              type="text"
              value=""
            />
            
          </div>
        </div>
        <button
          class="btn btn__danger landing__btn btn__medium btn__primary"
          type="button"
        >
          Submit
        </button>
        <div
          class="__react_component_tooltip place-right type-dark"
          data-id="tooltip"
          id="test"
        >
          <div
            class="landing__tooltip"
          >
            <p>
              For assistance,
            </p>
            <p
              class="landing__tooltip__contact"
            >
              please contact us:
            </p>
            <p>
              Monday - friday
            </p>
            <p>
              9am - 6pm est
            </p>
            <p>
              Saturday
            </p>
            <p
              class="landing__tooltip__contact"
            >
              9am - 5pm est
            </p>
            <p>
              clientservices@dvf.com
            </p>
            <p>
              888-472-2383
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
